feat: add query to iterate over all parameters in btcstaking

RafilxTenfen opened the pull request.

GAtom22 commented on proto/babylon/btcstaking/v1/query.proto:
```diff
@@ -345,3 +351,20 @@ message QueryLargestBtcReOrgResponse {
+// QueryParamsVersionsRequest is the request type for the
+// Query/ParamsVersions RPC method.
+message QueryParamsVersionsRequest {
+  // pagination defines an optional pagination for the request.
```
Nit, add annotations to avoid generating code that will not be used, e.g.:

```suggestion
  option (gogoproto.equal) = false;
  option (gogoproto.goproto_getters) = false;
  
  // pagination defines an optional pagination for the request.
```

GAtom22 commented on proto/babylon/btcstaking/v1/query.proto:
```diff
@@ -345,3 +351,20 @@ message QueryLargestBtcReOrgResponse {
+
+// QueryParamsVersionsResponse stores all the params with versions.
+message QueryParamsVersionsResponse {
+  // params holds all the params with version from this module.
```
ditto here
```suggestion
  option (gogoproto.equal) = false;
  option (gogoproto.goproto_getters) = false;
  
  // params holds all the params with version from this module.
```

GAtom22 reviewed (commented):
LGTM! Great work @RafilxTenfen!!

Left a small nit.
One question, do we want to add this as a cli command also? In such case, that is missing

RafilxTenfen commented:
> One question, do we want to add this as a cli command also? In such case, that is missing
Good idea, added the query to CLI and addressed the pr comments 

@GAtom22 Can you kindly review again?

GAtom22 commented on x/btcstaking/client/cli/query_params.go:
```diff
@@ -35,3 +35,29 @@ func CmdQueryParams() *cobra.Command {
+
+			queryClient := types.NewQueryClient(clientCtx)
+
+			res, err := queryClient.ParamsVersions(cmd.Context(), &types.QueryParamsVersionsRequest{})
```
Pagination missing on request. You can get it from the flags

```suggestion
                        pageReq, err := client.ReadPageRequest(cmd.Flags())
			if err != nil {
				return err
			}
			res, err := queryClient.ParamsVersions(cmd.Context(), &types.QueryParamsVersionsRequest{
			     Pagination: pageReq,
			})
```

RafilxTenfen replied:
good catch

GAtom22 reviewed (approved):
LGTM! :rocket:
